src/rpdk/core/project.py

```diff
@@ -499,6 +501,22 @@ def submit(
                         )
                     self._plugin.package(self, zip_file)
 
+                    # Plugin package method adds the plugin language and version info
+                    # to the metadata file. Update the file with cli version info and copy
+                    # to the zip artifact
+                    try:
+                        version_metadata = {}
+                        with open(CFN_METADATA_FILE_NAME, "r") as metadata_file:
+                            version_metadata = json.load(metadata_file)
+
+                        with open(CFN_METADATA_FILE_NAME, "w") as metadata_file:
+                            version_metadata["cli-version"] = __version__
+                            json.dump(version_metadata, metadata_file)
+
+                        zip_file.write(CFN_METADATA_FILE_NAME)
```

Comment:
if we are creating this then just using it to write to the zipfile, then removing want to use writestr to remove the initial opening of the file and removal of file after? https://docs.python.org/3/library/zipfile.html#zipfile.ZipFile.writestr

Reply:
Since the plugins and cli are writing the version info independently, and zip archives cannot be updated, I believe we have to either write to a file and then copy it over to the archive, or create a new archive after the plugins have finished writing the version metadata.

Reply:
so plugins write to a zip file, then we open it here and write to the same file? maybe im missing the plugin portion as it would be in a subsequent PR. I think a better approach would be to implement a get_version_information method in each plugin code, pass that back to the main class, then write once

Reply:
Sounds good, updated the PR. Corresponding Java plugin PR -> https://github.com/aws-cloudformation/cloudformation-cli-java-plugin/pull/349/files